src/main/java/org/myrobotlab/client/InProcessCli.java

```diff
@@ -85,15 +308,12 @@ public void stop() {
 
   public static void main(String[] args) {
     try {
-      // Logger logger = (Logger)
-      // LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
-      // logger.setLevel(Level.INFO);
 
       // Logger.getRootLogger().setLevel(Level.INFO);
 
       Runtime.getInstance();
 
-      InProcessCli client = new InProcessCli(System.in, System.out);
+      InProcessCli client = new InProcessCli(NameGenerator.getName(), "test", System.in, System.out);
```

Comment:
Not interested at the moment in outputting to stderr

Reply:
i'm mostly concerned about the scenario where stderr output buffer gets written and filled and causes strange behaviors in services (like we saw with dl4j & opencv)  perhaps this is no longer attached to an actual spawned process and as a result, we don't have to worry about stderr now?

Reply:
builder.redirectErrorStream(true)  is still there ... so stderr goes to stdout and stdout is consumed by a StreamGobbler.  So, not much has changed there.  The thing that changed was we do not open  process.getOutputStream()  (remote process's stdin from the Agent). 

We had challenges with the Agent's handles of remote process streams.
InProcessCli has a different purpose - it does not handle a remote stream, it simply reads from it own process stdin and writes to stdout (it currently does not write to stderr and I don't see any reason to at the moment).

**--interactive-mode** is only used when you want to run Runtime without the Agent **AND** want cli control through stdin (so basically the old way, but without the Agent). This can be useful during debugging, but --interactive-mode is never sent by the Agent, so the remote process's stdin is never open during regular runtime.  This solves the most recent unyielding problem of opencv 1.5.1 being stuck.

All remote control now is done through websockets.
The Agent starts a listening webgui on port 8887 (without starting the browser), then spawns a new instance of MRL with a unique ID (new NameGenerator thingy).  By default the MRL instance will try to connect to port 8887, when/if it does, the Agent accepts the connection and then will have control of the instance.

This offers considerable advantages over previous control, and is similar in design to Apache Storm's Supervisor / Worker architecture.

https://www.tutorialspoint.com/apache_storm/apache_storm_cluster_architecture.htm

Agent's by default will only listen on localhost:8887 but a node is a node is a node.  With this new addressing scheme and all the refactoring work I did on gateways and routing it will be possible to connect everything in a distributed mesh, and still be able to manage from a cental endpoint ...